Client/Frontend/Browser/HomePanel/BraveShieldStatsView.swift

```diff
@@ -103,9 +103,16 @@ class BraveShieldStatsView: UIView, Themeable {
                 text = Strings.ShieldsTimeStatsDays
             }
             
-            return "\(Int(counter))\(text)"
+            return toLocaleString(num: Int(counter)) + text
         }
     }
+    
+    private func toLocaleString(num: Int) -> String {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.numberStyle = NumberFormatter.Style.decimal
+        numberFormatter.locale = NSLocale.current
+        return numberFormatter.string(from: num as NSNumber) ?? ""
```

Comment:
To be more explicit, let's return `nil`, and then in usage locations we can fallback to `"0"` (making the return type `String?` here).